src/core/js/app.js

```diff
@@ -25,201 +16,33 @@ require([
+], function (Adapt, NavigationView) {
 
     // Append loading template and show
-    window.Handlebars = _.extend(require("handlebars"), window.Handlebars);
+    window.Handlebars = _.extend(require('handlebars'), window.Handlebars);
 
-    var template = Handlebars.templates['loading'];
-    $('body').append(template());
+    $('body').append(Handlebars.templates.loading());
 
-    Adapt.build = new BuildModel(null, {url: "adapt/js/build.min.js", reset:true});
 
-    // This function is called anytime a course object is loaded
-    // Once all course files are loaded trigger events and call Adapt.initialize
-    Adapt.checkDataIsLoaded = function(newLanguage) {
-        if (Adapt.contentObjects.models.length > 0 &&
-            Adapt.articles.models.length > 0 &&
-            Adapt.blocks.models.length > 0 &&
-            Adapt.components.models.length > 0 &&
-            Adapt.course.get('_id')) {
-
-            mapAdaptIdsToObjects();
-
-            Adapt.log.debug('Firing app:dataLoaded');
-
-            try {
-                Adapt.trigger('app:dataLoaded');// Triggered to setup model connections in AdaptModel.js
-            } catch(e) {
-                Adapt.log.error('Error during app:dataLoading trigger', e);
-            }
-
-            Adapt.setupMapping();
-
-            Adapt.wait.queue(function() {
-                triggerDataReady(newLanguage);
-            });
-
-        }
-    };
-
-    function triggerDataReady(newLanguage) {
-        if (newLanguage) {
-
-            Adapt.trigger('app:languageChanged', newLanguage);
-
-            _.defer(function() {
-                var startController = new StartController();
-                var hash = '#/';
-
-                if (startController.isEnabled()) {
-                    hash = startController.getStartHash(true);
-                }
-
-                Backbone.history.navigate(hash, { trigger: true, replace: true });
-            });
-        }
-
-        Adapt.log.debug('Firing app:dataReady');
-
-        try {
-            Adapt.trigger('app:dataReady');
-        } catch(e) {
-            Adapt.log.error('Error during app:dataReady trigger', e);
-        }
-
-        Adapt.wait.queue(triggerInitialize);
-
-    }
-
-    function triggerInitialize() {
-        Adapt.log.debug('Calling Adapt.initialize');
+    Adapt.data.whenReady().then(function triggerInit() {
+        Adapt.log.debug('Calling Adapt.init');
 
         addNavigationBar();
 
-        Adapt.initialize();
+        Adapt.init();
 
         Adapt.off('adaptCollection:dataLoaded courseModel:dataLoaded');
-    }
+    });
 
     function addNavigationBar() {
 
-        var adaptConfig = Adapt.course.get("_navigation");
+        var adaptConfig = Adapt.course.get('_navigation');
 
         if (adaptConfig && adaptConfig._isDefaultNavigationDisabled) {
```

Comment:
as far as I can see, this is the only reference to `_isDefaultNavigationDisabled` (and there's no documentation or schema entry for it)

is this something we actually need?

if not we could drop this, move L46 into `triggerInitialize` and remove the entire `addNavigationBar` function

Reply:
This is a precursor to getting rid of the navigation bar altogether and putting it in a separate plugin. Ideally we'd just document the property?